### PR TITLE
feat: enum support

### LINKED ISSUE #133: Handling of enums
Given an API specification like this:
```
"OrderByEnum": {
        "enum": [
          1,
          2
        ],
        "type": "integer",
        "description": "\r\n* **1** - Ascending\r\n* **2** - Descending\r\n\n\n1 = Ascending\n\n2 = Descending",
        "format": "int32",
        "x-enumNames": [
          "Ascending",
          "Descending"
        ]
      }
```
Currently, the generated type ends up being:
```
/**
 * * **1** - Ascending
 * * **2** - Descending
 *
 *
 * 1 = Ascending
 *
 * 2 = Descending
 *
 * @format int32
 */
export type OrderByEnum = 1 | 2;
```
Taking advantage of the x-enumNames, we could actually create a more desirable output like:
```
export enum OrderByEnum {
    Ascending = 1,
    Descending = 2,
}
```
This is something that NSwagStudio does, it's universally supported, e.g. https://github.com/unchase/Unchase.Swashbuckle.AspNetCore.Extensions search for AddEnumsWithValuesFixFilters.
The biggest problem (other than convenience) arises when you want to use the enum in the flag. In the previous code, with enum I could do:
```
const test = OrderByEnum.Ascending | OrderByEnum.Descending;
```
Which is valid, and also in many cases desirable, meanwhile with the current code:
```
const test:  OrderByEnum = 3;
```
Will not work because 3 is not allowed by the union type.

### PR DESCRIPTION
**Pull request by alan-oliv**
# Description

This is a new feature that enables outputting enums instead of string unions, even without open-api enum $ref in generated swagger schema. (Fixes #133)

## Type of change
- [x] New feature (non-breaking change which adds functionality)

# How to
in your defineConfig, you have to pass `useEnums: true`

image

# Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

# Support
- [x] String enums
- [x] Integer enums
- [x] Boolean enums

## String enums
Given a schema:
`
  "MyStringEnum": {
      "type": "string",
       "enum": ["ENABLED", "DISABLED"]
   }
`

the output will be:
`
export enum MyStringEnum {
  ENABLED = "ENABLED",
  DISABLED = "DISABLED",
}`

## Int enums
Given a schema:
`
  "MyIntEnum": {
        "type": "int",
        "enum": [0, 1, 2, 3, 4]
    }
`

the output will be:
`
export enum MyIntEnum {
  ZERO = 0,
  ONE = 1,
  TWO = 2,
  THREE = 3,
  FOUR = 4,
}`

## Boolean enums
Given a schema:
`
  "MyBooleanEnum": {
        "type": "boolean",
        "enum": [true, false]
    }
`

the output will be:
`
export enum MyBooleanEnum {
  True,
  False
}`

### CODE REVIEW

**Comment by shadowlik** [comment]
+1 to this feature

**Comment by davidsonsns** [comment]
+1

**Comment by DavidVeloso** [comment]
+1

File: plugins/typescript/src/utils/getEnumProperties.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  "thirty",
+  "forty",
+  "fifty",
+  "sixty",
+  "seventy",
+  "eighty",
+  "ninety",
+];
+
+export const convertNumberToWord = (n: number): string => {
```
Comment (fabien0102) [inline plugins/typescript/src/utils/getEnumProperties.ts:92]: Let’s add a dummy unit-test for this util, like a little loop to test, let’s say, the main/tricky cases 😉

**Review by fabien0102 — CHANGES_REQUESTED** [review]
Amazy pull-request! Thanks a lot for this contribution, let’s just add a little unit test on `convertNumberToWord` and we are good to go!

**Review by fabien0102 — APPROVED** [review]
Amazing!

**Pull request merged**